### ISSUE TITLE
[Feature request] updateNumericInputIcon

### ISSUE THREAD

**Issue by H4R81N63R** [dreamRs/shinyWidgets#278]
Hello. I am really enjoying using your package, excellent work and thank you so much for it.

I was wonder if it would be possible for you to implement an update function for the numericInputIcon() UI element, much like the updateNumericInput() for the numericInput() UI element in base Shiny.

The use case I have is that I am allowing the user to input the number of pixels in height and width for the image they want to export. The internal range I am using is between 100 and 10,000 pixels. However, the user does not get to see this (due to the restrictions of the UI design). As such, the user can enter any numeric value, positive or negative, in the input field as shown in this image,

![Screenshot from 2020-05-10 12-10-56](https://user-images.githubusercontent.com/9100674/81497716-15775000-92b8-11ea-9cc2-455577c5a807.png)

Now I can handle this programmatically at the server end, but the user does not get the feedback of the limitations. I have tried the sliders, but then I/the user loses out on accuracy down to single pixels.

I can do this with the Shiny built-in numericInput() and updateNumericInput() pair enclosed in observe({}) where if a user inputs a value out of internal range then it is immediately updated to the closest bound (upper or lower), but then I lose out on the icons.

In the shinyWidget package I can find the updateNumericRangeInput(), but not the updateNumericInputIcon() and I was wondering if you could implement this.

Cheers

**Comment by pvictor** [comment]
Hello,
Glad you like the package :)

My bad, this is not documented, you can actually use `updateNumericInput` to update value since `numericInputIcon()` and `numericInput` share their bindings. In the next release I'll make an alias for the sake of clarity.

Example:

```r
library(shiny)
library(shinyWidgets)

ui <- fluidPage(
  
  fluidRow(
    column(
      width = 6,
      numericInputIcon(
        inputId = "width",
        label = "Width",
        value = 500, 
        min = 100, 
        max = 10000,
        icon = list(icon("ruler-horizontal"), "px")
      ),
      verbatimTextOutput("res1"),
      actionButton("update", "Update to 2000")
    )
  )
)

server <- function(input, output, session) {
  
  output$res1 <- renderPrint(input$width)
  
  observeEvent(input$update, {
    updateNumericInput(session, "width", value = 2000)
  })

}

shinyApp(ui, server)
```

Bonus:
To show to user that the value is out of range, you can use this little javascript snippet:

```r
$(document).on("shiny:inputchanged", function(event) {
  if (event.inputType === "shiny.number") {
    if (event.value === null || event.value < 100 || event.value > 10000) {
      $("#" + event.name).parents(".form-group").addClass("has-error");
    } else {
      $("#" + event.name).parents(".form-group").removeClass("has-error");
    }
  }
});
```
Save this to a file and include it in your app with `includeScript("validate-number.js")` in your UI. It will affect all `numericInput` or `numericInputIcon`, you can make it specific to one by adding `event.name === "width"` in the first `if`.

Result will look like:

![image](https://user-images.githubusercontent.com/4415580/81532932-917ba180-9365-11ea-94c0-0375c6d52853.png)

Victor

**Comment by H4R81N63R** [comment]
Thank you very much for your reply and the solution. I had tried the updateNumericInput() earlier on a hunch, but in an oversight, I was also passing the icon information to it leading to errors and incompatibility. I have fixed it based on your reply and it is working great now!

Just brainstorming, but I wonder if it would be worthwhile to expand the alias to also allow a change in the icons if a developer wants to implement such a feature?
